arbnode/dataposter/storage_test.go

```diff
@@ -0,0 +1,256 @@
+package dataposter
+
+import (
+	"context"
+	"path"
+	"strconv"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/google/go-cmp/cmp"
+	"github.com/offchainlabs/nitro/arbnode/dataposter/leveldb"
+	"github.com/offchainlabs/nitro/arbnode/dataposter/redis"
+	"github.com/offchainlabs/nitro/arbnode/dataposter/slice"
+	"github.com/offchainlabs/nitro/util/arbmath"
+	"github.com/offchainlabs/nitro/util/redisutil"
+	"github.com/offchainlabs/nitro/util/signature"
+)
+
+func newLevelDBStorage[Item any](t *testing.T) *leveldb.Storage[Item] {
+	t.Helper()
+	db, err := rawdb.NewLevelDBDatabase(path.Join(t.TempDir(), "level.db"), 0, 0, "default", false)
```

Comment:
Let's use `rawdb.NewMemoryDatabase()` here instead for simplicity

Reply:
It's a test for leveldb, I think it makes sense to actually use leveldb here. Besides it doesn't complicate anything really.

Reply:
Makes sense